### PR TITLE
Bug 1561617 - Support Fennec on esr68 rules

### PR DESCRIPTION
**Pull request by JohanLorenzo**
@rvandermeulen defined the scheme at https://bugzilla.mozilla.org/show_bug.cgi?id=1548655#c0

### CODE REVIEW

File: mozilla_version/gecko.py [hunk 1]
```diff
@@ -343,6 +359,20 @@ class FennecVersion(_VersionWithEdgeCases):
...
+        """Ensure attributes are sane all together."""
+        # Versions matching 68.XbN are expected since bug 1523402.
+        if (
+            self.major_number == 68 and self.is_beta and self.patch_number is None and
+            self.minor_number > 0
+        ):
+            return
+
+        if self.major_number >= 69:
+            raise PatternNotMatchedError(self, pattern='Last Fennec version is 68')
```
Comment (Callek) [inline mozilla_version/gecko.py]: This one could be worrisome if we use this for GeckoView anywhere, but I'm not going to block on it since the error message should be clear what is wrong.

File: mozilla_version/test/test_gecko.py [hunk 2]
```diff
@@ -330,6 +330,36 @@ def test_fennec_version_supports_released_edge_cases(version_string):
...
+    ('68.1.1b2', True),
+
+    ('69.0a1', True),
+    ('69.0b3', True),
+    ('69.0', True),
+    ('69.0.1', True),
+
+    ('70.0', True),
+))
+def test_fennec_version_ends_at_68(version_string, raises):
```
Comment (Callek) [inline mozilla_version/test/test_gecko.py]: I'm a fan of using a method like https://docs.pytest.org/en/latest/example/parametrize.html#parametrizing-conditional-raising for this, rather than `if .. else` to call the same function... but not-blocking

Comment (JohanLorenzo) [reply mozilla_version/test/test_gecko.py]: Oooooh TIL! I've used this way because that's what we do on scriptworker and most *script. I'll change this. Thanks!

File: mozilla_version/gecko.py (hunk 1 above)
Comment (JohanLorenzo) [reply mozilla_version/gecko.py]: Good call. Geckoview should use MAJOR.MINOR.BUILD_ID which is parsed by https://github.com/mozilla-releng/mozilla-version/blob/e5400f31f7001bd48fb6e17626905147dd4c17d7/mozilla_version/maven.py#L13. I searched mozilla-central, but the python doesn't compute anything before https://searchfox.org/mozilla-central/rev/7e158713cf5a8514fa8161dd4a239737b05da64d/mobile/android/geckoview/build.gradle#52 happens.

In short, I don't expect any breakage, if there's one, we should probably address it by expanding MavenVersion.
